v2/oauth/common-actions.mdx

```diff
@@ -0,0 +1,37 @@
+---
+title: Common Actions
+hide_title: true
+---
+
+# Common Actions
+
+## Working with Scopes
+
+The allowed scopes are set during the creation process of an **OAuth2 Client**.
+
+By default, our **OAuth2** implementation adds the following built-in scopes:
+- `email`: adds the `email` and `email_verified` claims into the Id Token and the User Info.
+- `phoneNumber`: adds the `phoneNumber` and `phoneNumber_verified` claims into the Id Token and the User Info
+- `roles`: adds the roles claim into the Id Token and Access Token.
+   - This will contain the roles returned by `getRolesForUser`
+   - This only works if the `UserRoles` recipe is initialized
+- `permissions`: adds the `permissions` claim into the Id Token and Access Token.
+   - This will contain the list of permissions obtained by concatenating the result of `getPermissionsForRole` for all roles returned by `getRolesForUser`
+   - This only works if the `UserRoles` recipe is initialized
+
+### Requesting Specific Scopes
+
+The client can request specific scopes by adding `scope` query param to the Authorization URL.
+The requested scopes have to be a subset of what is allowed for the client, otherwise the authentication requst will fail.
+By default all scopes are granted to the client.
+
+### Overriding granted scopes
+
+If you want to manually modify the list of scopes that are granted to the client during the authentication flow you can do this by using overrides.
+You will have to add a custom implementation for `getConsentRequest` and change the value it returns for `requestedScope`.
```

Comment:
please give code examples here

Reply:
I can help with code snippets next week, but please add a list of snippets you'd like help with and their descriptions.